cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1469,6 +1469,14 @@ macro(build_gtest)
     set(GTEST_CMAKE_ARGS ${GTEST_CMAKE_ARGS} "-DCMAKE_MACOSX_RPATH:BOOL=ON")
   endif()
 
+  if(CMAKE_GENERATOR STREQUAL "Xcode")
+    # Xcode projects support multi-configuration builds.  This forces the gtest build
+    # to use the same output directory as a single-configuration Makefile driven build.
+    set(GTEST_CMAKE_ARGS
+        ${GTEST_CMAKE_ARGS} "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=${_GTEST_LIBRARY_DIR}"
+        "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_${CMAKE_BUILD_TYPE}=${_GTEST_RUNTIME_DIR}")
```

Comment:
Could you use `list(APPEND GTEST_CMAKE_ARGS ...)`?

Reply:
Good tip, thanks @kou for reviewing these changes!